joblib/hashing.py

```diff
@@ -129,8 +129,13 @@ def save_global(self, obj, name=None, pack=struct.pack):
     dispatch[type(pickle.dump)] = save_global
 
     def _batch_setitems(self, items):
-        # forces order of keys in dict to ensure consistent hash
-        Pickler._batch_setitems(self, iter(sorted(items)))
+        # forces order of keys in dict to ensure consistent hash.
+        # keys are ordered using their hash.
+        if PY3:
+            Pickler._batch_setitems(self, sorted(items,
+                                                 key=lambda k: hash(k[0])))
+        else:
+            Pickler._batch_setitems(self, iter(sorted(items)))
```

Comment:
@ogrisel, this was related to some test failing. I wanted to keep the same test cases to work. See `expected_dict` values in https://github.com/joblib/joblib/blob/master/joblib/test/test_hashing.py#L368 and in https://github.com/joblib/joblib/blob/master/joblib/test/test_hashing.py#L415
If I use the same sorting code for both python 2 and 3, this breaks one of the expected values for py2.